### PR TITLE
fix: rust keyword conflict

### LINKED ISSUE #343: [Bug]  sol macro doesn't handle solidity idents that are rust keywords
### Component

json-abi, sol! macro, syn-solidity

### What version of Alloy are you on?

_No response_

### Operating System

None

### Describe the bug

image
image
https://etherscan.io/address/0xdef1c0ded9bec7f1a1670819833240f027b25eff#code

### PR DESCRIPTION
**Pull request by Will-Smith11**
## Motivation
Fixes #343

## Solution
Take all unique rust keywords and enforce them to be `RawIdents` in `SolIdent`

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/syn-solidity/src/ident/mod.rs [hunk 1]
```diff
@@ -71,7 +84,13 @@ impl Parse for SolIdent {
...
 impl ToTokens for SolIdent {
     fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
-        self.0.to_tokens(tokens);
+        let str = self.0.to_string();
+        if RUST_KEYWORD_SET_DIFFERENCE.contains(&str.as_str()) {
+            Ident::new_raw(&str, self.span())
+        } else {
+            Ident::new(&str, self.span())
+        }
+        .to_tokens(tokens)
```
[the remark this answers is not in the document]
Comment (Will-Smith11) [reply crates/syn-solidity/src/ident/mod.rs:93]: If the logic is moved to on construction it breaks the system

[the remark this answers is not in the document]
Comment (Will-Smith11) [reply crates/syn-solidity/src/ident/mod.rs:93]: > If the logic is moved to on construction it breaks the system

was able to fix it. Some weird logic with a From impl was causing a `not a valid identifier error`

**Comment by Will-Smith11** [comment]
@DaniPopes  Not sure why the CI is failing, everything on my local is working when i run the same commands. Any idea?

**Comment by DaniPopes** [comment]
You have to rebase to main. Tests run on merge with main, not on the latest commit

**Pull request merged**